### PR TITLE
refactor: resolve warnings in RCTLoginButton and FBLoginButtonManager

### PR DESCRIPTION
**Pull request by rnike**
Resolved some warnings in RCTLoginButton and FBLoginButtonManager

* nullable warnings
* variable should be marked as `final` warning
* variable never accessed warning

## Test Plain

1. `yarn example:install && yarn example:devcopy && yarn example:install && yarn example:android`
2. Press on the login button
3. The alert should show after the login process, it means the onChange is working properly, no matter login is successful or not, I tested `onCancel` and `onError`.

### CODE REVIEW

File: android/src/main/java/com/facebook/reactnative/androidsdk/FBLoginButtonManager.java [hunk 1]
```diff
@@ -49,17 +49,28 @@ public String getName() {
...
         return new RCTLoginButton(context, mActivityEventListener.getCallbackManager());
-
     }
 
     @ReactProp(name = "loginBehaviorAndroid")
     public void setLoginBehavior(RCTLoginButton loginButton, @Nullable String loginBehavior) {
-        loginButton.setLoginBehavior(LoginBehavior.valueOf(loginBehavior.toUpperCase()));
+        LoginBehavior behavior = LoginBehavior.NATIVE_WITH_FALLBACK;
+
+        if(loginBehavior != null){
```
Comment (mikehardy) [inline android/src/main/java/com/facebook/reactnative/androidsdk/FBLoginButtonManager.java:58]:
```suggestion
        if (loginBehavior != null) {
```

File: android/src/main/java/com/facebook/reactnative/androidsdk/FBLoginButtonManager.java [hunk 2]
```diff
@@ -49,17 +49,28 @@ public String getName() {
...
+
+        loginButton.setLoginBehavior(behavior);
     }
 
     @ReactProp(name = "defaultAudience")
     public void setDefaultAudience(RCTLoginButton loginButton, @Nullable String defaultAudience) {
-        loginButton.setDefaultAudience(DefaultAudience.valueOf(defaultAudience.toUpperCase()));
+        DefaultAudience audience = DefaultAudience.FRIENDS;
+
+        if(defaultAudience != null){
```
Comment (mikehardy) [inline android/src/main/java/com/facebook/reactnative/androidsdk/FBLoginButtonManager.java:69]:
```suggestion
        if (defaultAudience != null) {
```

File: android/src/main/java/com/facebook/reactnative/androidsdk/RCTLoginButton.java [hunk 3]
```diff
@@ -63,7 +61,7 @@ protected void onCurrentAccessTokenChanged(
                     WritableMap event = Arguments.createMap();
                     event.putString("type", "logoutFinished");
                     ReactContext context = (ReactContext) getContext();
-                    context.getJSModule(RCTEventEmitter.class).receiveEvent(
+                    context.getJSModule(ReactEventEmitter.class).receiveEvent(
```
Comment (mikehardy) [inline android/src/main/java/com/facebook/reactnative/androidsdk/RCTLoginButton.java:64]: Out of curiosity, why not the style that is documented? Using: RCTDeviceEventEmitter?

https://reactnative.dev/docs/native-modules-android#sending-events-to-javascript

**Review by mikehardy — CHANGES_REQUESTED** [review]
two tiny spacing issues (I should really integrate this package I made so that it gets checked pre-commit - the goal is uniformity more than any particular style - https://github.com/invertase/nodejs-google-java-format) and a question on deviation from documented event emitter style

Thanks for posting this though - setting up my test rig for fbsdk-next right now before the family wakes up and hopefully can get that one reviewed and in shortly

File: android/src/main/java/com/facebook/reactnative/androidsdk/RCTLoginButton.java (hunk 3 above)
Comment (rnike) [reply android/src/main/java/com/facebook/reactnative/androidsdk/RCTLoginButton.java:64]: Didn't see that, I think you are right it should be using `RCTDeviceEventEmitter` instead, I'll change and test it tomorrow. Turning this pr to draft for now until it's ready.

Comment (rnike) [reply android/src/main/java/com/facebook/reactnative/androidsdk/RCTLoginButton.java]: After all, this should be `RCTEventEmitter` according to https://reactnative.dev/docs/native-components-android#events, not sure why it's marked deprecated on AndroidStudio, but currently it's a won't fix.

**Review by thebergamo — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
LGTM - merging in to beta branch, I just generated a release there with the AEMLogger changes, once that release (beta2) is done I'll make a beta3 with this one

**Comment by mikehardy** [comment]
Ah, with "refactor" tags on the commits no release will actually trigger for these. They are merged in to the beta branch though. @thebergamo @rnike - I personally have been really impressed with your contributions here Mike - are you interested in a more formal maintainership role so you can more directly help instead of always relying on my or Marcos? Marcos - are you interested in delegating more authority to Mike? In my experience, the more maintainers the better, even though there is never a formal commitment since it's open source of course. Cheers :-)

**Comment by thebergamo** [comment]
Yeah, specially that lately I'm having almost 0 time to keep it up :(

So I'm 100% on @rnike becoming a more formal "Maintainer" hehe

**Comment by rnike** [comment]
@mikehardy @thebergamo Thanks for the invitation, that would be my honor to have the place. I'm not really good at English(not a native English user/speaker), still will need help especially for the documentation part. Anyway, practice makes better, I'll take it.

**Comment by mikehardy** [comment]
Your English is fantastic and you should know that, from a native speaker. But I'm a US English speaker that moved to Spanish speaking Ecuador and I do understand that it is always a struggle right? It seems like there is always a mistake when I re-read my written Spanish, so I totally understand. Great to hear you are interested!

**Comment by thebergamo** [comment]
No doubt your English is great :)
Also not native, but as you said practice helps to improve and it's a daily exercise hehe

**Pull request merged**